src/Aspire.Hosting/ResourceBuilderExtensions.cs

```diff
@@ -283,6 +283,22 @@ public static IResourceBuilder<T> WithManifestPublishingCallback<T>(this IResour
         return builder.WithAnnotation(new ManifestPublishingCallbackAnnotation(callback), ResourceAnnotationMutationBehavior.Replace);
     }
 
+    /// <summary>
+    /// Registers an async callback which is invoked when publishing is performed for the app model.
+    /// </summary>
+    /// <typeparam name="T">The resource type.</typeparam>
+    /// <param name="builder">The resource builder.</param>
+    /// <param name="callback">Callback method which takes a <see cref="DefaultPublishingContext"/> which can be used to publish assets.</param>
+    /// <returns></returns>
+    public static IResourceBuilder<T> WithDefaultPublishingCallback<T>(this IResourceBuilder<T> builder, Func<DefaultPublishingContext, Task> callback) where T : IResource
```

Comment:
I'm not sure about the name default. I'm wondering whether we should just have PublishingContext, WithPublishingCallback etc.

Reply:
Done